### PR TITLE
fix instances of: c++11 range-loop might detach Qt container warnings

### PR DESCRIPTION
**Pull request by mgallien**
analyzed via clazy

Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

File: src/gui/folderman.cpp [hunk 1]
```diff
@@ -594,7 +597,8 @@ Folder *FolderMan::folder(const QString &alias)
 
 void FolderMan::scheduleAllFolders()
 {
-    for (Folder *f : _folderMap.values()) {
+    const auto folderMapValues = _folderMap.values();
+    for (Folder *f : folderMapValues) {
```
Comment (claucambra) [inline src/gui/folderman.cpp:601]: might be out of scope of this PR, but `auto`

Comment (claucambra) [reply src/gui/folderman.cpp:601]: also, I don't think single letter variables are ever a good idea (though that might be slightly out of scope of this PR too)

File: src/gui/folderman.cpp [hunk 2]
```diff
@@ -720,7 +724,8 @@ void FolderMan::slotAccountStateChanged()
     if (accountState->isConnected()) {
         qCInfo(lcFolderMan) << "Account" << accountName << "connected, scheduling its folders";
 
-        for (Folder *f : _folderMap.values()) {
+        const auto folderMapValues = _folderMap.values();
+        for (Folder *f : folderMapValues) {
```
Comment (claucambra) [inline src/gui/folderman.cpp:728]: `auto`

also a single letter variable

File: src/gui/folderman.cpp [hunk 3]
```diff
@@ -1202,7 +1207,8 @@ QStringList FolderMan::findFileInLocalFolders(const QString &relPath, const Acco
     if (!serverPath.startsWith('/'))
         serverPath.prepend('/');
 
-    for (Folder *folder : this->map().values()) {
+    const auto mapValues = map().values();
+    for (Folder *folder : mapValues) {
```
Comment (claucambra) [inline src/gui/folderman.cpp:1211]: `auto` and single letter variable

File: src/gui/folderman.cpp [hunk 4]
```diff
@@ -1400,7 +1406,8 @@ void FolderMan::slotWipeFolderForAccount(AccountState *accountState)
 
 void FolderMan::setDirtyProxy()
 {
-    for (const Folder *f : _folderMap.values()) {
+    const auto folderMapValues = _folderMap.values();
+    for (const Folder *f : folderMapValues) {
```
Comment (claucambra) [inline src/gui/folderman.cpp:1410]: `auto` and single letter variable

File: src/gui/folderman.cpp [hunk 5]
```diff
@@ -1414,7 +1421,8 @@ void FolderMan::setDirtyProxy()
 
 void FolderMan::setDirtyNetworkLimits()
 {
-    for (Folder *f : _folderMap.values()) {
+    const auto folderMapValues = _folderMap.values();
+    for (Folder *f : folderMapValues) {
```
Comment (claucambra) [inline src/gui/folderman.cpp:1425]: `auto` and single letter variable

File: src/gui/tray/usermodel.cpp [hunk 6]
```diff
@@ -220,7 +220,8 @@ void User::slotReceivedPushActivity(Account *account)
 
 void User::slotCheckExpiredActivities()
 {
-    for (const Activity &activity : _activityModel->errorsList()) {
+    const auto errorsList = _activityModel->errorsList();
+    for (const Activity &activity : errorsList) {
```
Comment (claucambra) [inline src/gui/tray/usermodel.cpp:224]: `auto` could be used here

File: test/testallfilesdeleted.cpp [hunk 7]
```diff
@@ -70,8 +70,10 @@ private slots:
             });
 
         auto &modifier = deleteOnRemote ? fakeFolder.remoteModifier() : fakeFolder.localModifier();
-        for (const auto &s : fakeFolder.currentRemoteState().children.keys())
+        const auto childrenKeys = fakeFolder.currentRemoteState().children.keys();
+        for (const auto &s : childrenKeys) {
```
Comment (claucambra) [inline test/testallfilesdeleted.cpp:74]: this is also a single letter variable

File: test/testallfilesdeleted.cpp [hunk 8]
```diff
@@ -110,8 +112,10 @@ private slots:
...
```
Comment (claucambra) [inline test/testallfilesdeleted.cpp:116]: single letter variable

File: test/testallfilesdeleted.cpp [hunk 9]
```diff
@@ -139,8 +143,10 @@ private slots:
             [&] { QVERIFY(false); });
         QVERIFY(fakeFolder.syncOnce());
 
-        for (const auto &s : fakeFolder.currentRemoteState().children.keys())
+        const auto childrenKeys = fakeFolder.currentRemoteState().children.keys();
+        for (const auto &s : childrenKeys) {
```
Comment (claucambra) [inline test/testallfilesdeleted.cpp:147]: single letter variable

File: src/3rdparty/kirigami/wheelhandler.cpp [hunk 10]
```diff
@@ -71,7 +71,8 @@ bool GlobalWheelFilter::eventFilter(QObject *watched, QEvent *event)
         bool shouldBlock = false;
         bool shouldScrollFlickable = false;
 
-        for (auto *handler : m_handlersForItem.values(item)) {
+        const auto handlerItems = m_handlersForItem.values(item);
+        for (auto *handler : handlerItems) {
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/3rdparty/kirigami/wheelhandler.cpp:75]: yes
but should we not keep this focused on the specific warning of detaching Qt containers ?

[the remark this answers is not in the document]
Comment (claucambra) [reply src/3rdparty/kirigami/wheelhandler.cpp:75]: sure, no worries

File: test/testallfilesdeleted.cpp (hunk 7 above)
Comment (allexzander) [reply test/testallfilesdeleted.cpp:74]: I agree with Claudio, especially on this. `s` is misleading, it does not stand for `childrenKeys`. Something like `chKey` would be much more readable.

File: test/testallfilesdeleted.cpp (hunk 8 above)
Comment (allexzander) [reply test/testallfilesdeleted.cpp:116]: Same

File: test/testsyncconflict.cpp [hunk 11]
```diff
@@ -79,7 +79,8 @@ private slots:
         QVERIFY(fakeFolder.syncOnce());
 
         // Verify that the conflict names don't have the user name
-        for (const auto &name : findConflicts(fakeFolder.currentLocalState().children["A"])) {
+        const auto conflicts = findConflicts(fakeFolder.currentLocalState().children["A"]);
+        for (const auto &name : conflicts) {
```
Comment (allexzander) [inline test/testsyncconflict.cpp:83]: @mgallien `name` -> `conflict`?

**Review by allexzander — APPROVED** [review]
Approved, but, kindly see if you can fix our nitpicks on naming.

**Pull request merged**